### DISCUSSION TITLE
How do you negatively delay events?

### DISCUSSION THREAD

**Discussion by Alkamist** [free-audio/clap#320] [Q&A] [answered]
I'm trying to make a plugin that can pre-delay MIDI events and I can't seem to figure out how to do it. I am using the latency extension to set the latency of the plugin, which I can see is working in my DAW, but I am unsure how to actually utilize the latency to negatively delay the MIDI.

**Comment by Alkamist** [comment]
To clarify my problem, when I set the latency with the latency extension, the plugin reports that there is latency to my DAW, but I am not seeing any events ahead of time. If I immediately send events when I see them in the process function, they are not pre-delayed.

**Reply by defiantnerd** [reply]
Which host do you use? Usually, the events are addressed within the current process block, the audible result is expected to be rendered in a later sample or even in a later process call.

**Reply by Alkamist** [reply]
I am using Reaper. It is entirely possible I am doing something wrong though. So you are saying that if I set the plugin to have latency, and then within a process call immediately pass on input events to output events, they are supposed to show up negatively delayed? That is what I would expect to happen and it doesn't seem like it is.

**Comment by Alkamist** [comment]
I seem to have fixed it but I'm a bit confused. I was thinking since the time field in the event header is uint32_t, that it would only be positive, but if I send negative times and underflow that value they are showing up properly. Is this expected behavior?

**Reply by defiantnerd** [reply, marked as answer]
No, I don‘t think so. Probably there is a bug in Reaper. PDC is hard sometimes. :)

**Reply by Alkamist** [reply]
This seems to have been a bug which I informed the Reaper devs about. It looks like it has been addressed in the dev releases.
